packages/assets-controllers/src/AssetsContractController.test.ts

```diff
@@ -56,31 +61,39 @@ const TEST_ACCOUNT_PUBLIC_ADDRESS =
  */
 async function setupAssetContractControllers({
   options,
-  useNetworkControllerProvider,
+  useNetworkControllerProvider = false,
   infuraProjectId = '341eacb578dd44a1a049cbc5f6fd4035',
 }: {
-  options?: Partial<ConstructorParameters<typeof AssetsContractController>[0]>;
+  options?: Partial<
+    Omit<ConstructorParameters<typeof AssetsContractController>[0], 'messenger'>
+  >;
   useNetworkControllerProvider?: boolean;
   infuraProjectId?: string;
 } = {}) {
   const networkClientConfiguration = {
     type: NetworkClientType.Infura,
-    network: 'mainnet',
+    network: NetworkType.mainnet,
     infuraProjectId,
     chainId: BUILT_IN_NETWORKS.mainnet.chainId,
     ticker: BUILT_IN_NETWORKS.mainnet.ticker,
   } as const;
   let provider: Provider;
 
-  const messenger: NetworkControllerMessenger =
-    new ControllerMessenger().getRestricted({
+  const controllerMessenger = new ControllerMessenger<
+    | NetworkControllerActions
+    | AssetsContractControllerActions
+    | AssetsContractAllowedActions,
+    | NetworkControllerEvents
+    | AssetsContractControllerEvents
+    | AssetsContractAllowedEvents
+  >();
```

Comment:
I believe we can do this more easily without having to export `AllowedActions` and `AllowedEvents` now. A while back I added a couple of utility types for extracting the action and event types from a controller messenger type. So what if we say this instead:

``` suggestion
  const controllerMessenger = new ControllerMessenger<
    | ExtractAvailableAction<AssetsContractControllerMessenger>
    | NetworkControllerActions,
    | ExtractAvailableEvent<AssetsContractControllerMessenger>
    | NetworkControllerEvents
  >();